examples/shader/post_processing.rs

```diff
@@ -0,0 +1,321 @@
+        .add_startup_system(setup)
+        .add_system(main_pass_cube_rotator_system);
+
+    let render_app = app.sub_app_mut(RenderApp);
+    let driver = PostProcessPassCameraDriver::new(&mut render_app.world);
+
+    let mut graph = render_app.world.resource_mut::<RenderGraph>();
+
+    // Add a node for the post processing pass.
+    graph.add_node(POST_PROCESS_PASS_DRIVER, driver);
+
+    // The post process pass's dependencies include those of the main pass.
+    graph
+        .add_node_edge(node::MAIN_PASS_DEPENDENCIES, POST_PROCESS_PASS_DRIVER)
+        .unwrap();
+
+    // Insert the post process pass node: CLEAR_PASS_DRIVER -> MAIN_PASS_DRIVER -> POST_PROCESS_PASS_DRIVER
+    graph
+        .add_node_edge(POST_PROCESS_PASS_DRIVER, node::MAIN_PASS_DRIVER)
+        .unwrap();
+    app.run();
+}
+
+/// A node for the `PostProcessingPassCamera` that runs `draw_2d_graph` with this camera.
+struct PostProcessPassCameraDriver {
+    query: QueryState<Entity, With<PostProcessingPassCamera>>,
+}
+
+impl PostProcessPassCameraDriver {
+    pub fn new(render_world: &mut World) -> Self {
+        Self {
+            query: QueryState::new(render_world),
+        }
+    }
+}
+impl Node for PostProcessPassCameraDriver {
+    fn update(&mut self, world: &mut World) {
+        self.query.update_archetypes(world);
+    }
+
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        _render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        for camera in self.query.iter_manual(world) {
+            graph.run_sub_graph(draw_2d_graph::NAME, vec![SlotValue::Entity(camera)])?;
+        }
+        Ok(())
+    }
+}
+
+/// Marks the Main pass cube (rendered to a texture.)
+#[derive(Component)]
+struct MainPassCube;
+
+fn setup(
+    mut commands: Commands,
+    mut windows: ResMut<Windows>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut post_processing_materials: ResMut<Assets<PostProcessingMaterial>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut images: ResMut<Assets<Image>>,
+    mut clear_colors: ResMut<RenderTargetClearColors>,
+) {
+    let window = windows.get_primary_mut().unwrap();
+    let size = Extent3d {
+        width: window.physical_width(),
+        height: window.physical_height(),
+        ..default()
+    };
+
+    // This is the texture that will be rendered to.
+    let mut image = Image {
+        texture_descriptor: TextureDescriptor {
+            label: None,
+            size,
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING
+                | TextureUsages::COPY_DST
+                | TextureUsages::RENDER_ATTACHMENT,
+        },
+        ..default()
+    };
+
+    // fill image.data with zeroes
+    image.resize(size);
+
+    let image_handle = images.add(image);
```

Comment:
All that code should not be needed if the rendering pipeling correctly sets its output such as https://github.com/bevyengine/bevy/pull/3430/files#diff-ee49b813badd948f80ad07ecd71b9cce1d766335e6923fa24694d85b86f7efbfR78 ; I think I should remove all that code and make it work.

Reply:
this is still a bit of an issue, but clearly a lesser one now, so I'll consider it resolved.